org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/internal/javadoc/CoreJavadocAccessImpl.java

```diff
@@ -975,15 +975,31 @@ else if (isSnippet) {
 
 		if (isReturn)
 			fBuf.append(JavaDocMessages.JavadocContentAccess2_returns_post);
-		if (isCode || (isLink && addCodeTagOnLink()))
-			fBuf.append("</code>"); //$NON-NLS-1$
+		if (isCode || (isLink && addCodeTagOnLink())) {
+			if (isCode) {
+				ASTNode sibling = getNextSiblingElement((TagElement)node.getParent(), node);
```

Comment:
Do you have a test scenario where getNextSibling is null and so you don't add the closing </code> tag?  It should be added. 

FWIW: I did work on pre code tag support and nested inline tags in an inline return tag.

Minor nit: the code indentation is off.

Reply:
I do not have a valid test scenario where `getNextSibling` is null, but I can mimic one.

```
/**
	 * Performs:
	 * <pre>{@code
	 *    for (String s : strings) {
	 *        if (s.equals(value)) {
	 *            return \u0030;
	 *        }
	 *        if (s.startsWith(value)) {
	 *            return 1;
	 *        }
	 *    }
	 *    return -1;
	 * }
	 */
```
Here, I removed the closing `pre` tag. In this case, getNextSibling becomes null.

You are right, it is better to add an `else` for the `if (sibling != null)` condition

Reply:
I tried a test with no <pre> tags and it works as expected.